src/commands/TruffleCommands.ts

```diff
@@ -188,12 +188,35 @@ export namespace TruffleCommands {
     Telemetry.sendEvent('TruffleCommands.writeBytecodeToBuffer.commandFinished');
   }
 
+  /**
+   * Creates a new contract file named `NewContract.sol`.
+   *
+   * If `folderUri` is provided, the new contract will be created in that folder.
+   * It **must** represent a folder URI.
+   *
+   * Otherwise, it uses {@link getTruffleWorkspace} to select the
+   * Truffle workspace to place the new contract.
+   *
+   * Once the new contract file has been created,
+   * the _Contract Explorer_ view will be refreshed.
+   *
+   * @param folderUri if provided, the `Uri` to place the newly created contract.
+   */
   export async function createContract(folderUri?: Uri): Promise<void> {
-    const uri = await getTruffleWorkspace(folderUri);
-    const workspace = uri.workspace;
-    const contractDirectory = getPathByPlatform(workspace);
+    let folderPath: string;
+
+    if (folderUri === undefined) {
+      const truffleWorkspace = await getTruffleWorkspace();
```

Comment:
this is good overall, is there room to think about making this more generic in terms of workspace? as part of the SDK Core Commands the concept of a workspace is in there v's a hard function?

Reply:
Maybe we could reuse some parts of the Truffle SDK into the, let's say, HardHat SDK. But first I guess we need to define some sort of common denominator functionality to add the SDK Core Commands. Then see how we can implement it.